lib/rules/font-weight-notation/index.js

```diff
@@ -124,45 +125,39 @@ const rule = (primary, secondaryOptions, context) => {
 
 			if (primary === 'numeric') {
 				if (!isNumbery(lowerWeightValue) && fontWeightNonNumericKeywords.has(lowerWeightValue)) {
-					if (context.fix) {
-						const numericValue = KEYWORD_TO_NUMERIC.get(lowerWeightValue);
+					const numericValue = NAMED_TO_NUMERIC.get(lowerWeightValue);
 
+					if (context.fix) {
 						if (numericValue) {
 							weightValueNode.value = numericValue;
 
 							return true;
 						}
 					}
 
-					complain(messages.expected('numeric'), weightValueNode);
+					const msg = numericValue
+						? messages.expectedWithActual(weightValue, numericValue)
+						: messages.expected('numeric');
+
+					complain(msg, weightValueNode);
 
 					return true;
 				}
 			}
 
 			if (primary === 'named-where-possible') {
-				if (isNumbery(lowerWeightValue) && NUMERIC_TO_KEYWORD.has(lowerWeightValue)) {
-					if (context.fix) {
-						const keyword = NUMERIC_TO_KEYWORD.get(lowerWeightValue);
+				if (isNumbery(lowerWeightValue) && NUMERIC_TO_NAMED.has(lowerWeightValue)) {
+					const namedValue = NUMERIC_TO_NAMED.get(lowerWeightValue);
 
-						if (keyword) {
-							weightValueNode.value = keyword;
-						}
+					assertString(namedValue);
```

Comment:
[note] Use `assertString` for less `if` branches. `NUMERIC_TO_NAMED.has() === true` ensures the return value of ` NUMERIC_TO_NAMED.get()` is a string.